lib/main/AT32F43x/middlewares/i2c_application_library/i2c_application.c

```diff
@@ -1069,8 +1107,12 @@ i2c_status_type i2c_memory_address_send(i2c_handle_type* hi2c, i2c_mem_address_w
 i2c_status_type i2c_memory_write(i2c_handle_type* hi2c, i2c_mem_address_width_type mem_address_width, uint16_t address, uint16_t mem_address, uint8_t* pdata, uint16_t size, uint32_t timeout)
 {
   /* initialization parameters */
-  hi2c->pbuff = pdata;
-  hi2c->pcount = size + mem_address_width;
+  hi2c->mode   = I2C_MA_TX;
+
+  // Spoon feeding data without a preceding register write
+  hi2c->step = I2C_STEP_DATA;
+  hi2c->pbuff[I2C_STEP_DATA] = pdata;
+  hi2c->pcount[I2C_STEP_DATA] = size + mem_address_width;
```

Comment:
The spoon feeding code is popping up several times. Maybe it makes sense to introduce a static function that handles all instances of this?